src/pudl/analysis/allocate_net_gen.py

```diff
@@ -196,43 +197,169 @@
+        """Allocate net gen from gen_fuel to generator/energy_source_code level."""
+        gf, bf, gen, bga, gens = select_input_data(
+            gf=gf, bf=bf, gen=gen, bga=bga, gens=gens
+        )
+        return allocate_gen_fuel_by_generator_energy_source(
+            gf=gf,
+            bf=bf,
+            gen=gen,
+            bga=bga,
+            gens=gens,
+            freq=freq,
+            debug=context.op_config["debug"],
+        )
+
+    @asset(
+        name=f"generation_fuel_by_generator_{agg_freqs[freq]}_eia923",
+        ins={
+            "net_gen_fuel_alloc": AssetIn(
+                key=f"generation_fuel_by_generator_energy_source_{agg_freqs[freq]}_eia923"
+            ),
+            "pu": AssetIn(key="denorm_plants_utilities_eia"),
+            "bga": AssetIn(key="boiler_generator_assn_eia860"),
+        },
+        io_manager_key=io_manager_key,
+        compute_kind="Python",
+    )
+    def gen_fuel_by_gen(
+        net_gen_fuel_alloc: pd.DataFrame, pu: pd.DataFrame, bga: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Aggregate gen fuel data columns to generators."""
+        return (
+            # aggregate the gen/pm/fuel records back to generator records
+            agg_by_generator(
+                net_gen_fuel_alloc=net_gen_fuel_alloc,
+                sum_cols=DATA_COLUMNS,
+            )
+            # make the output resemble denorm_generation_eia923:
+            .pipe(pudl.output.eia923.denorm_by_gen, pu=pu, bga=bga)
+        )
+
+    @asset(
+        name=f"generation_fuel_by_generator_energy_source_owner_{agg_freqs[freq]}_eia923",
+        ins={
+            "net_gen_fuel_alloc": AssetIn(
+                key=f"generation_fuel_by_generator_energy_source_{agg_freqs[freq]}_eia923"
+            ),
+            "gens": AssetIn(key="denorm_generators_eia"),
+            "own_eia860": AssetIn(key="denorm_ownership_eia860"),
+        },
+        io_manager_key=io_manager_key,
+        compute_kind="Python",
+    )
+    def gen_fuel_by_gen_esc_owner(
+        net_gen_fuel_alloc: pd.DataFrame, gens: pd.DataFrame, own_eia860: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Aggregate gen fuel data columns to generator owners."""
+        return pudl.analysis.plant_parts_eia.MakeMegaGenTbl().scale_by_ownership(
+            gens_mega=pudl.helpers.date_merge(
+                left=net_gen_fuel_alloc,
+                right=gens[IDX_GENS + ["utility_id_eia", "capacity_mw"]],
+                left_date_col="report_date",
+                right_date_col="report_date",
+                new_date_col="report_date",
+                on=["plant_id_eia", "generator_id"],
+                date_on=["year"],
+                how="left",
+                report_at_start=True,
+                validate="m:1",
+            ),
+            own_eia860=own_eia860,
+            scale_cols=DATA_COLUMNS + ["capacity_mw"],
+            validate="m:m",  # m:m because there are multiple generators in gen_pm_fuel
+        )
```

Comment:
in a similar fashion to the other assets in here, does it make sense to have the actual work happening inside the factory insides of calling a version of `scale_allocated_net_gen_by_ownership`? if you are going to leave it in here i think you should at least migrate the more fleshed out documentation of that step into this function w/in a factory

Reply:
also now i'm noticing that you did not remove `scale_allocated_net_gen_by_ownership` so you should certainly do that or rely on it.

Reply:
Ah whoops you're right. It would be more analogous to use the outside function. I'll update it.